test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/SraIdentityResolutionUsingPluginsTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.CompletableFuture;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.core.SdkPlugin;
+import software.amazon.awssdk.core.SdkServiceClientConfiguration;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.identity.spi.ResolveIdentityRequest;
+import software.amazon.awssdk.services.protocolquery.ProtocolQueryClient;
+import software.amazon.awssdk.services.protocolquery.ProtocolQueryServiceClientConfiguration;
+import software.amazon.awssdk.utils.Validate;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SraIdentityResolutionUsingPluginsTest {
+
+    @Mock
+    private AwsCredentialsProvider credentialsProvider;
+
+    @Test
+    public void testIdentityBasedPluginsResolutionIsUsedAndNotAnotherIdentityResolution() {
+        when(credentialsProvider.identityType()).thenReturn(AwsCredentialsIdentity.class);
+        when(credentialsProvider.resolveIdentity(any(ResolveIdentityRequest.class)))
+            .thenReturn(CompletableFuture.completedFuture(AwsBasicCredentials.create("akid1", "skid2")));
+
+        ProtocolQueryClient syncClient = ProtocolQueryClient
+            .builder()
+            .addPlugin(new TestPlugin(credentialsProvider))
+            // Below is necessary to create the test case where, addCredentialsToExecutionAttributes was getting called before
+            .overrideConfiguration(ClientOverrideConfiguration.builder().build())
+            .build();
+
+        try {
+            syncClient.allTypes(builder -> {
+            });
+        } catch (Exception expected) {
+            expected.printStackTrace(System.out);
+        }
```

Comment:
Can we adapt this test (and the other one...) to throw an exception before we send the request, and not print a stack trace that just make the logs longer. Searching for `boom` (!) in the codebase gives a few different options.

Reply:
> Can we adapt this test (and the other one...) to throw an exception before we send the request,

Sure, not sure how to do that, I can look around but if you have any pointers that will be helpful.

> and not print a stack trace that just make the logs longer.

For sure, I will remove it, it was not really meant to part of this change.

Reply:
I like this pattern:

```
SdkHttpClient mockClient = mock(SdkHttpClient.class);
when(mockClient.prepareRequest(any())).thenThrow(new RuntimeException("boom"));

ProtocolQueryClient syncClient = ProtocolQueryClient
    .builder()
    .httpClient(mockClient)
    .build();

assertThatThrownBy(() -> syncClient.allTypes(r -> {})).hasMessageContaining("boom");
```

Reply:
Sounds good, I will create a follow up PR to adjust the tests accordingly.